crates/mun_codegen_macros/src/lib.rs

```diff
@@ -213,8 +332,370 @@ pub fn as_value_derive(input: TokenStream) -> TokenStream {
         Data::Union(_) => {
             unimplemented!("#[derive(AsValue)] is not defined for unions!");
         }
-        Data::Enum(_) => {
-            unimplemented!("#[derive(AsValue)] is not defined for enums!");
+        Data::Enum(enum_data) => {
+            const SUPPORTED_TAG_SIZES: &[&str] =
+                &["u8", "u16", "u32", "u64", "i8", "i16", "i32", "i64"];
+
+            let repr_ty = derive_input.attrs.iter().find_map(|a| {
+                a.parse_meta().map_or(None, |m| {
+                    if let syn::Meta::List(list) = m {
+                        let op = list
+                            .path
+                            .segments
+                            .iter()
+                            .next()
+                            .map(|s| s.ident.to_string());
+
+                        if op == Some("repr".to_string()) {
+                            return list.nested.iter().next().and_then(|n| {
+                                if let syn::NestedMeta::Meta(m) = n {
+                                    m.path().segments.iter().next().map(|s| s.ident.clone())
+                                } else {
+                                    None
+                                }
+                            });
+                        }
+                    }
+
+                    None
+                })
+            });
+
+            let repr_ty = if let Some(ident) = repr_ty {
+                let repr_ty = ident.to_string();
+                if !SUPPORTED_TAG_SIZES.contains(&repr_ty.as_str()) {
+                    eprintln!(
+                        "`repr({})` is not supported by the `AsValue` macro.",
+                        repr_ty
+                    );
+                }
+
+                quote! {
+                    #ident
+                }
+            } else {
+                // Default to u32
+                quote! {
+                    u32
+                }
+            };
+
+            if enum_data.variants.is_empty() {
```

Comment:
Should we add a todo on the backlog for this?

Reply:
Does it add value to support an `enum` without variants? E.g. this is no legal in Rust:

```rust
enum Foo {} // legal

fn main() {
    let foo = Foo {}; // illegal: error[E0574]: expected struct, variant or union type, found enum `Foo`
    println!("Hello, world!");
}
```

Reply:
I see thats fine then :)